ft_strnstr.c

```diff
@@ -0,0 +1,55 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   ft_strnstr.c                                       :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: hbalan <hbalan@student.42.fr>              +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/01/19 12:13:17 by hbalan            #+#    #+#             */
+/*   Updated: 2023/01/19 12:13:19 by hbalan           ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <string.h>
+#include <stdio.h>
+
+char	*ft_strnstr(const char *str, const char *to_find, size_t len)
+{
+	size_t	i;
+	size_t	j;
+
+	i = 0;
+	if (!str && !len)
+		return (0);
+	if (to_find[0] == '\0' || to_find == str)
+		return ((char *)str);
+	while (str[i] != '\0')
+	{
+		j = 0;
+		while (str[i + j] == to_find[j] && (i + j) < len)
+		{
+			if (str[i + j] == '\0' && to_find[j] == '\0')
+				return ((char *)&str[i]);
+				//return ((char *)str + i);
+			j++;
+		}
+		if (to_find[j] == '\0')
+			// return ((char *)(str + i));
+			return ((char *)&str[i]);
+		i++;
+	}
+	return (0);
+}
+
+/*int	main(void)
+{
+ char *largestring = "Foo Bar Baz";
+ char *smallstring = "Bar";
+char *ptr;
+
+//this returns null cause only first 4 characters of largestring were searched
+//ptr = strnstr(largestring, smallstring, 4);
+
+ptr = ft_strnstr(largestring, smallstring, 7);
+printf("%s\n", ptr);
+}*/
```

Comment:
Works as a charm, but I still can't fully understand it, I will make a mental note to myself about it, however, tested on like 15 cases, looks awersome!